osu.Game/Overlays/Settings/RulesetSettingsSubsection.cs

```diff
@@ -25,9 +28,9 @@ protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnl
         {
             dependencies = new DependencyContainer(base.CreateChildDependencies(parent));
 
-            var config = dependencies.Get<RulesetConfigCache>().GetConfigFor(ruleset);
-            if (config != null)
-                dependencies.Cache(config);
+            Config = dependencies.Get<RulesetConfigCache>().GetConfigFor(ruleset);
+            if (Config != null)
+                dependencies.CacheAs(Config);
```

Comment:
Wait why was this changed to a CacheAs?

Reply:
If not it will be cached as an `IRulesetConfigManager`, which is less useful to children (would need a cast at each usage).

Reply:
No, it's cached as an `IRulesetConfigManager` if you use `CacheAs`. `CacheAs` uses the type passed into the generic parameter (`IRulesetConfigManager` in this case).

Reply:
so it does. that method name is a bit confusing..